library/choose_disk.py

```diff
@@ -0,0 +1,651 @@
+
+            current_physical_disk = physical_disks[physical_disk]
+            match_count = 0
+            # Checking what features are matching
+            for feature in current_lookup:
+                if feature not in current_physical_disk:
+                    continue
+
+                # Assign left and right operands
+                right = current_lookup[feature]
+                left = current_physical_disk[feature]
+
+                def evaluate_operator(left, right):
+                    # Default comparing operator is equal
+                    operator = "equal"
+
+                    # Test if we have another operator in the right operand
+                    arguments = _REGEXP.search(right)
+                    if arguments:
+                            new_operator = arguments.group(1)
+
+                            # Some operators are aliases to more complex commands.
+                            # Let's make the substition in place and restart with it
+                            alias = get_alias(new_operator, arguments.group(2), arguments.group(3))
+                            if alias:
+                                return evaluate_operator(left, alias)
+
+                            # Check if the associated function exists
+                            if new_operator in OPERATORS:
+                                # and assign operands with the new values
+                                operator = new_operator
+                                right = arguments.group(2)
+                                new_arguments = _REGEXP.search(right)
+                                if new_arguments:
+                                    # Don't forget to evaluate the two sides of the expression
+                                    # Typical case when we shall compare a 'value' with : 'and( gt(x), lt(y) )'
+                                    # The looking value always stays to the 'left' part of the expression
+                                    new_right = arguments.group(3)
+                                    return OPERATORS[operator](evaluate_operator(left, right), evaluate_operator(left, new_right))
+                                    #           and           (                  value,gt(x)),                   value,lt(y)     )
+                            else:
+                                fatal("Unsupported '%s' operator in : %s" % (new_operator, right), module)
+                    return OPERATORS[operator](left, right)
+
+                # Let's check if (left <operator> right) is True meaning the match is done
+                if evaluate_operator(left, right):
+                    logger.debug("  %s : match  %s %s", physical_disk, left, right)
+                    match_count = match_count + 1
+                    continue
+                else:
+                    # comparaison is false meaning devices doesn't match
+                    logger.debug("  %s : no match  %s %s", physical_disk, left, right)
```

Comment:
is there some relation between the visual blocks and the logging statements? For example, if there is a log line that has two leading spaces vs. another one that has none (or just 1), would that mean some kind of hierarchy? The reason why I was asking for normalizing is because it is easier for parsing. If there is a need for better visuals maybe create a different logger?